app/src/main/java/com/marcelorcorrea/falae/fragment/ViewPagerItemFragment.kt

```diff
@@ -99,11 +99,8 @@ class ViewPagerItemFragment : Fragment() {
             frameLayout.background = drawable
         }
         frameLayout.setOnClickListener {
-            var itemSelected = item
-            if (SharedPreferencesUtils.getBoolean(SettingsFragment.SCAN_MODE, context)) {
-                itemSelected = mItems[currentItemSelectedFromScan]
-            }
-            onItemClicked(itemSelected)
+            val currentScannedItem = getScannedItem() ?: item
```

Comment:
maybe change this constant name to something like currentItem since scan_mode might not be enabled.